oak-run/src/main/java/org/apache/jackrabbit/oak/indexversion/IndexVersionOperation.java

```diff
@@ -86,42 +92,65 @@ public static List<IndexVersionOperation> generateIndexVersionOperationList(Node
 
             // for rest indexes except active index
-            for (IndexName indexNameObject : reverseSortedIndexNameList) {
-                String indexName = indexNameObject.getNodeName();
-                NodeState indexNode = indexDefParentNode.getChildNode(PathUtils.getName(indexName));
-                IndexVersionOperation indexVersionOperation = new IndexVersionOperation(indexNameObject);
-                // if active index not long enough, NOOP for all indexes
-                if (isActiveIndexLongEnough) {
-                    if (indexNameObject.getProductVersion() == activeProductVersion && indexNameObject.getCustomerVersion() == 0) {
-                        indexVersionOperation.setOperation(Operation.DELETE_HIDDEN_AND_DISABLE);
-                    } else {
-                        // the check hidden oak mount logic only works when passing through the proper composite store
-                        if (isHiddenOakMountExists(indexNode)) {
-                            LOG.info("Found hidden oak mount node for: '{}', disable it but no index definition deletion", indexName);
+                for (IndexName indexNameObject : reverseSortedIndexNameList) {
+                    String indexName = indexNameObject.getNodeName();
+                    NodeState indexNode = indexDefParentNode.getChildNode(PathUtils.getName(indexName));
+                    IndexVersionOperation indexVersionOperation = new IndexVersionOperation(indexNameObject);
+                    // if active index not long enough, NOOP for all indexes
+                    if (isActiveIndexLongEnough) {
+                        if (indexNameObject.getProductVersion() == activeProductVersion && indexNameObject.getCustomerVersion() == 0) {
                             indexVersionOperation.setOperation(Operation.DELETE_HIDDEN_AND_DISABLE);
-                        } else {
-                            indexVersionOperation.setOperation(Operation.DELETE);
+                        } else if (indexNameObject.getProductVersion() <= activeProductVersion ) {
+                            // the check hidden oak mount logic only works when passing through the proper composite store
+                            if (isHiddenOakMountExists(indexNode)) {
+                                LOG.info("Found hidden oak mount node for: '{}', disable it but no index definition deletion", indexName);
+                                indexVersionOperation.setOperation(Operation.DELETE_HIDDEN_AND_DISABLE);
+                            } else {
+                                indexVersionOperation.setOperation(Operation.DELETE);
+                            }
                         }
+                        // if the index product version is larger than active index, leave it as is
+                        // for instance: if there is active damAssetLucene-7, the inactive damAssetLucene-8 will be leave there as is
```

Comment:
Shouldn't we add an "else" case where, where we also log this case (info level), and set the operation? (Operation.NOOP). I know, Operation.NOOP is the default (and that's good), so setting this to NOOP is a no-op :-) But when reading this code here, that's not obvious.

Reply:
@thomasmueller makes sense, let me refine this a bit to have a log here

Reply:
@thomasmueller just push a commit to reflect this, can you check it again? thanks!